bundle/src/main/java/com/adobe/acs/commons/dispatcher/impl/PermissionSensitiveCacheServlet.java

```diff
@@ -0,0 +1,121 @@
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.dispatcher.impl;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.felix.scr.annotations.*;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static com.adobe.acs.commons.dispatcher.impl.PermissionSensitiveCacheServlet.REMOVEABLE_EXTENSIONS;
+
+@Component(
+        label = "ACS AEM Commons - Permission Sensitive Cache Servlet",
+        description = "Servlet that checks if the current sessions has access to a cached object",
+        metatype = true, immediate = true
+)
+@Properties({
+        @Property(
+                label = "File extensions to be removed from the URI",
+                name = REMOVEABLE_EXTENSIONS,
+                description = "Extensions that can be removed from the URI. For example, .html",
+                cardinality = Integer.MAX_VALUE,
+                value = {"html"}
+        ),
+        @Property(
+                name = "sling.servlet.paths",
+                cardinality = Integer.MAX_VALUE,
+                label = "Sling Servlet Paths",
+                description = "Paths that this servlet will resolve to"
+        )
+})
+@Service
+public class PermissionSensitiveCacheServlet extends SlingSafeMethodsServlet {
+
+    private final Logger log = LoggerFactory.getLogger(PermissionSensitiveCacheServlet.class);
+    public static final String REMOVEABLE_EXTENSIONS = "removable-extensions";
+
+    private List<String> removableExtensions = new ArrayList<String>();
+
+    @Activate
+    public final void activate(final Map<String, String> config) {
+
+        String[] propertyArray = PropertiesUtil.toStringArray(config.get(REMOVEABLE_EXTENSIONS),
+                new String[]{"html"});
+        removableExtensions = Arrays.asList( propertyArray );
+    }
+
+    public void doHead(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        doHeadRequest(request, response);
+    }
+
+    public void doHeadRequest(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        try{
+            //retrieve the requested URL
+            String requestUri = request.getParameter( "uri" );
+            log.debug( "Request URI {}", requestUri );
+
+            if(StringUtils.isNotEmpty( requestUri ) ){
+
+                String uri = getUri( requestUri );
+
+                log.info( "Checking permissions for {}", uri );
+
+                Resource requestedResource = request.getResourceResolver().getResource( uri );
```

Comment:
Would it be better to use `rr.resolve(request, uri)` ? It seems like this would simplify this servlet and handle other corner cases:

* selectors
* use of /etc/map
* vanities (cant recall if getResource() will work for them, but resolve def does)

It comes at the cost of a single call to resolve (which seems like a reasonable trade-off).